### PR TITLE
Address unfulfilled requirements for RAQUASI88

### PR DESCRIPTION
**Pull request by rzumer**
This PR addresses the following checklist items that were improperly implemented:
- Unidentified game provides some descriptor [typically filename] to unknown game dialog
- HitCounts persist with save state (RA_OnSaveState, RA_OnLoadState)
- Full-window overlay can be navigated (RA_UpdateRenderOverlay>pInput)

Also includes the following changes:
- Provides basic tape support
    - A disk in drive 1 and a load tape should not be allowed to be inserted simultaneously in Hardcore mode
    - When resetting to enter Hardcore mode, any loaded disk will be prioritized over a tape, and the latter will be ejected
- Improves version management to separate RA-enabled and RA-disabled QUASI88 builds
- Automatically sets the speed to 100% when initializing with slowdown under Hardcore mode

I should note that while media swapping and other manipulations are blocked as much as possible without sacrificing required functionality, it is a computer system with built-in BASIC interpreter, so it is probably technically possible to manipulate memory by writing or side-loading a different program than intended. That said, it is hard enough to manipulate memory precisely through these methods that I doubt it will be an issue.

### CODE REVIEW

File: RAQUASI88/src/WIN32/retroachievements.cpp [hunk 1]
```diff
@@ -146,7 +152,7 @@ void RA_InitShared()
 static HDC main_hdc;
 void RA_InitUI()
 {
-    RA_Init(g_hWnd, /* RA_Quasi88 */ 9, Q_VERSION);
+    RA_Init(g_hWnd, RA_QUASI88, Q_VERSION);
```
[the remark this answers is not in the document]
Comment (rzumer) [reply RAQUASI88/src/WIN32/retroachievements.cpp]: I'm not sure how it makes a difference, as the updated version is required anyway, while the hardcoded 9 is opaque and will silently fail without an updated `RAIntegration` build.

[the remark this answers is not in the document]
Comment (Jamiras) [reply RAQUASI88/src/WIN32/retroachievements.cpp]: It's bookkeeping. We can update the RAIntegration submodule to master now, or wait for an official release and use that tag for the submodule update. Without the submodule update, this doesn't compile.

[the remark this answers is not in the document]
Comment (rzumer) [reply RAQUASI88/src/WIN32/retroachievements.cpp]: If updating the submodule with this PR is OK, then I prefer that.

File: RAQUASI88/src/WIN32/retroachievements.cpp [hunk 2]
```diff
@@ -120,9 +120,15 @@ void RebuildMenu()
 
 void GetEstimatedGameTitle(char* sNameOut)
 {
+    const int BUF_SIZE = 64;
+
     if (loaded_title[0] != NULL)
     {
-        sNameOut = (char*)loaded_title;
+        memcpy(sNameOut, loaded_title, BUF_SIZE);
```
[the remark this answers is not in the document]
Comment (rzumer) [reply RAQUASI88/src/WIN32/retroachievements.cpp]: Done.

**Comment by rzumer** [comment]
>I'm not sure if it's something in this PR or if I just missed it yesterday, but the dialog asking you to discard changes appears twice when changing the disk image.

Fixed.

**Pull request merged**